lib/auth/webauthn/login.go

```diff
@@ -0,0 +1,106 @@
+package webauthn
+
+import (
+	"context"
+
+	"github.com/duo-labs/webauthn/protocol"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+
+	wan "github.com/duo-labs/webauthn/webauthn"
+	wantypes "github.com/gravitational/teleport/api/types/webauthn"
+)
+
+// CredentialAssertion is the payload sent to authenticators to initiate login.
+type CredentialAssertion = protocol.CredentialAssertion
```

Comment:
What do the type aliases buy you here?

Reply:
They don't tie the rest of Teleport do duo-labs/webauthn, plus they allow future shenanigans if we want to extend the structs. For now I'm mostly playing it safe.

Reply:
But they don't really isolate you from duo-labs/webauthn because it's just an alias for that type, so any updates or new APIs to the upstream package will "bleed into" this package.

I've typically only seen type aliases used as a temporary solution during large refactoring, not as a permanent feature of the codebase.

If you really want to limit coupling to the duo-labs library then I'd recommend your own type that embeds `protocol.CredentialAssertion` or at least `type CredentialAssertion protocol.CredentialAssertion`, which would allow you to put your own methods on the type.

Reply:
Changed aliases to type definitions, good call there.

I think we're OK with the bleed given that updates are controlled via go.mod (and the bleed is likely to be desirable).